### ISSUE TITLE
extend test framework functions

### ISSUE THREAD

**Issue by pinheadmz** [bitcoin-dev-project/warnet#36]
anything pertaining to node must be extended into docker, examples:

`assert_debug_log()`
`restart_node()`
`add_nodes()`

also classes like `P2PInterface()` should be able to run locally but interact with deployed nodes

**Comment by willcl-ark** [comment]
Is this still an issue?

**Comment by pinheadmz** [comment]
yes a contributor trying to write a scenario using these or many other Test Framework methods is gonna have a bad time :-(

**Comment by Nosh-thee-techy** [comment]
Hii, is this still an issue?

**Comment by pinheadmz** [comment]
> Hii, is this still an issue?

Yes, do you understand what needs to be done and how it would benefit the warnet project and bitcoin more generally?
